feature-libs/user/profile/components/register/register-component.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import {
+  AuthService,
+  Command,
+  CommandService, Title,
+  User,
+  UserActions
+} from '@spartacus/core';
+import { UserProfileConnector, UserProfileService } from '@spartacus/user/profile/core';
+import { UserSignUp } from '@spartacus/user/profile/root';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+@Injectable()
+export class RegisterComponentService {
+  protected registerCommand: Command<{ user: UserSignUp }, User> =
+    this.command.create(({ user }) =>
+      this.userConnector.register(user).pipe(
+        tap(() => {
+          // this is a compatibility mechanism only, to make anonymous consents
+          // management work properly in transitional period (when we move logic
+          // to separate libraries)
+          this.store.dispatch(new UserActions.RegisterUserSuccess());
+        })
+      )
+    );
+
+  constructor(
+    protected userProfile: UserProfileService,
+    protected userConnector: UserProfileConnector,
+    protected authService: AuthService,
+    protected command: CommandService,
+    protected store: Store
+  ) {}
+
+  /**
+   * Register a new user.
+   *
+   * @param submitFormData as UserRegisterFormData
+   */
+  register(user: UserSignUp): Observable<User> {
+    return this.registerCommand.execute({ user });
+  }
+
+  /**
+   * Returns titles that can be used for the user profiles.
+   */
+  getTitles(): Observable<Title[]> {
+    return this.userProfile.getTitles();
+  }
```

Comment:
Instead of copy-pasting the implementation (and code comments like `// this is a compatibility mechanism only...` 😅 ) of the `UserRegisterService`, let's just delegate the calls to this service (to be precise: to it's facade `UserRegisterFacade`)

See the bottom right of the following chart https://github.com/SAP/spartacus/pull/15785#discussion_r884857549



```suggestion
export class RegisterComponentService {
  constructor(
    protected userRegisterFacade: UserRegisterFacade
  ) {}

  /**
   * Register a new user.
   *
   * @param submitFormData as UserRegisterFormData
   */
  register(user: UserSignUp): Observable<User> {
    return this.userRegisterFacade.register(user);
  }

  /**
   * Returns titles that can be used for the user profiles.
   */
  getTitles(): Observable<Title[]> {
    return this.userRegisterFacade.getTitles();
  }
```

Reply:
agreed, thanks!